diff --git a/Cargo.lock b/Cargo.lock
index 614dee5e70..793934b464 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -866,7 +866,7 @@
 [[package]]
 name = "cosmic-config"
 version = "0.1.0"
-source = "git+https://github.com/pop-os/libcosmic/#8a0e74b189f53dae9b8001c6fa5cf1820b3ececb"
+source = "git+https://github.com/pop-os/libcosmic/#b2b6d90fb63bac9e032f95e1523eef226b886f02"
 dependencies = [
  "atomicwrites",
  "calloop 0.14.2",
@@ -885,7 +885,7 @@
 [[package]]
 name = "cosmic-config-derive"
 version = "0.1.0"
-source = "git+https://github.com/pop-os/libcosmic/#8a0e74b189f53dae9b8001c6fa5cf1820b3ececb"
+source = "git+https://github.com/pop-os/libcosmic/#b2b6d90fb63bac9e032f95e1523eef226b886f02"
 dependencies = [
  "quote",
  "syn 1.0.109",
@@ -921,7 +921,7 @@
 [[package]]
 name = "cosmic-settings-config"
 version = "0.1.0"
-source = "git+https://github.com/pop-os/cosmic-settings-daemon#e2aa1056900d6f8c9c7555c0401aa7c99281eb06"
+source = "git+https://github.com/sefodopo/cosmic-settings-daemon?branch=keymap#6412c3ae99d50f09468c11eb16ba6cf2787da19f"
 dependencies = [
  "cosmic-config",
  "ron",
@@ -2242,7 +2242,7 @@
 [[package]]
 name = "iced_core"
 version = "0.14.0-dev"
-source = "git+https://github.com/pop-os/libcosmic/#8a0e74b189f53dae9b8001c6fa5cf1820b3ececb"
+source = "git+https://github.com/pop-os/libcosmic/#b2b6d90fb63bac9e032f95e1523eef226b886f02"
 dependencies = [
  "bitflags 2.8.0",
  "bytes",
@@ -2265,7 +2265,7 @@
 [[package]]
 name = "iced_futures"
 version = "0.14.0-dev"
-source = "git+https://github.com/pop-os/libcosmic/#8a0e74b189f53dae9b8001c6fa5cf1820b3ececb"
+source = "git+https://github.com/pop-os/libcosmic/#b2b6d90fb63bac9e032f95e1523eef226b886f02"
 dependencies = [
  "futures",
  "iced_core",
@@ -5960,7 +5960,7 @@
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "cf221c93e13a30d793f7645a0e7762c55d169dbb0a49671918a2319d289b10bb"
 dependencies = [
- "windows-sys 0.48.0",
+ "windows-sys 0.59.0",
 ]
 
 [[package]]
diff --git a/Cargo.toml b/Cargo.toml
index 7811025d44..f077d4868f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,48 +6,64 @@
 version = "0.1.0"
 
 [workspace]
-members = [
-  "cosmic-comp-config",
-]
+members = ["cosmic-comp-config"]
 
 [dependencies]
-anyhow = {version = "1.0.51", features = ["backtrace"]}
+anyhow = { version = "1.0.51", features = ["backtrace"] }
 bitflags = "2.4"
 bytemuck = "1.12"
-calloop = {version = "0.14.1", features = ["executor"]}
-cosmic-comp-config = {path = "cosmic-comp-config"}
-cosmic-config = {git = "https://github.com/pop-os/libcosmic/", features = ["calloop", "macro"]}
-cosmic-protocols = {git = "https://github.com/pop-os/cosmic-protocols", rev = "e706814", default-features = false, features = ["server"]}
-cosmic-settings-config = { git = "https://github.com/pop-os/cosmic-settings-daemon" }
+calloop = { version = "0.14.1", features = ["executor"] }
+cosmic-comp-config = { path = "cosmic-comp-config" }
+cosmic-config = { git = "https://github.com/pop-os/libcosmic/", features = [
+  "calloop",
+  "macro",
+] }
+cosmic-protocols = { git = "https://github.com/pop-os/cosmic-protocols", rev = "e706814", default-features = false, features = [
+  "server",
+] }
+cosmic-settings-config = { git = "https://github.com/sefodopo/cosmic-settings-daemon", branch = "keymap" }
 libdisplay-info = "0.2.0"
-egui = {version = "0.30.0", optional = true}
-egui_plot = {version = "0.30.0", optional = true}
-i18n-embed = {version = "0.14", features = ["fluent-system", "desktop-requester"]}
+egui = { version = "0.30.0", optional = true }
+egui_plot = { version = "0.30.0", optional = true }
+i18n-embed = { version = "0.14", features = [
+  "fluent-system",
+  "desktop-requester",
+] }
 i18n-embed-fl = "0.8"
-iced_tiny_skia = {git = "https://github.com/pop-os/libcosmic/"}
+iced_tiny_skia = { git = "https://github.com/pop-os/libcosmic/" }
 indexmap = "2.0"
 keyframe = "1.1.1"
 lazy_static = "1.4.0"
 libc = "0.2.149"
-libcosmic = {git = "https://github.com/pop-os/libcosmic/", default-features = false}
-libsystemd = {version = "0.7", optional = true}
-log-panics = {version = "2", features = ["with-backtrace"]}
+libcosmic = { git = "https://github.com/pop-os/libcosmic/", default-features = false }
+libsystemd = { version = "0.7", optional = true }
+log-panics = { version = "2", features = ["with-backtrace"] }
 once_cell = "1.18.0"
 ordered-float = "4.0"
 png = "0.17.5"
 regex = "1"
 ron = "0.9.0-alpha.0"
-rust-embed = {version = "8.0", features = ["debug-embed"]}
+rust-embed = { version = "8.0", features = ["debug-embed"] }
 sanitize-filename = "0.5.0"
 sendfd = "0.4.1"
-serde = {version = "1", features = ["derive"]}
+serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 thiserror = "1.0.26"
-time = {version = "0.3.30", features = ["macros", "formatting", "local-offset"]}
+time = { version = "0.3.30", features = [
+  "macros",
+  "formatting",
+  "local-offset",
+] }
 tiny-skia = "0.11"
-tracing = { version = "0.1.37", features = ["max_level_debug", "release_max_level_info"] }
+tracing = { version = "0.1.37", features = [
+  "max_level_debug",
+  "release_max_level_info",
+] }
 tracing-journald = "0.3.0"
-tracing-subscriber = {version = "0.3.16", features = ["env-filter", "tracing-log"]}
+tracing-subscriber = { version = "0.3.16", features = [
+  "env-filter",
+  "tracing-log",
+] }
 wayland-backend = "0.3.3"
 wayland-scanner = "0.31.1"
 xcursor = "0.3.3"
